controllers/managedocs_controller.go

```diff
@@ -175,14 +175,23 @@ func (r *ManagedOCSReconciler) generateStorageCluster(
 	if reconcileStrategy == v1.ReconcileStrategyStrict {
 		// Get an instance of the desired state
 		desired := utils.ObjectFromTemplate(templates.StorageClusterTemplate, r.Scheme).(*ocsv1.StorageCluster)
-
 		if err := r.setStorageClusterParamsFromSecret(desired); err != nil {
 			return err
 		}
 		// Override storage cluster spec with desired spec from the template.
 		// We do not replace meta or status on purpose
 
-		sc.Spec = desired.Spec
+		if sc.Spec.StorageDeviceSets == nil {
+			sc.Spec = desired.Spec
+		} else {
+			if sc.Spec.StorageDeviceSets[0].Count <= desired.Spec.StorageDeviceSets[0].Count {
+				sc.Spec = desired.Spec
+				r.Log.Info("Actual storage cluster parameter", "actualCount", sc.Spec.StorageDeviceSets[0].Count)
+			} else {
```

Comment:
In this case, you are just ignoring all other reconcile changes by not applying `desired.Spec` `to sc.Spec`. But this is not the desired behavior, other changes (not related to the sds count) should still be reflected